src/animated_sprite.c

```diff
@@ -33,10 +34,23 @@ void animated_sprite_tick(AnimatedSprite *anim, float anim_rate) {
 }
 
 void animated_sprite_draw(AnimatedSprite *anim, Position pos, Rect screen_rect) {
+	format_set_render_mode(anim->sprite, false);
+
 	if (is_intersecting(new_rect(pos, anim->size), screen_rect)) {
-		rdp_sync(SYNC_PIPE);
-		rdp_load_texture_stride(0, 0, MIRROR_DISABLED, anim->sprite, anim->_current_offset);
-		rdp_draw_sprite(0, pos.x - anim->render_offset.x, pos.y - anim->render_offset.y,
-						MIRROR_DISABLED);
+		surface_t spritesheet_surface = sprite_get_pixels(anim->sprite);
+
+		int tex_width = anim->sprite->width / anim->sprite->hslices;
```

Comment:
Does it make sense to use `anim->size` here instead? Not sure if we would have a situation where the size of a slice would be different from its size.
Although, if it does, should we change the `is_intersecting` above to use this instead? Or maybe not even receive the size and just calculate it as here on `init`?

Reply:
I was planning on completely overhauling the animation system in a separate PR to incorporate some of the testing I've been doing. I think all of those are good points, I'm just thinking that there might be situations where an animation might contain frames of different sizes or at least widths to save RDRAM usage

Reply:
We can leave that this way then and come back later.